slick.dataview.js

```diff
@@ -14,6 +14,25 @@
     }
   });
 
+  /**
+   * Polyfill for Map to support old browsers but
+   * benefit of the Map speed in modern browsers.
+   */
+  var SlickMap = 'Map' in window ? Map : function SlickMap() {
```

Comment:
nice polyfill, I think you move this code into the [slick.core.js](https://github.com/6pac/SlickGrid/blob/master/slick.core.js) so that it be used by other files, the core has these kind of util classes/fns so I think it's better there... and also expose it like the others and then you would call it with the dot instead `new Slick.Map()` (you can copy the structure of `Slick.Event` or the other ones in there)

The rest of the code looks quite clean, maybe just 1 other thing I could say is perhaps adding comments to some of the functions to explain what they do might be beneficial for future refactoring or understanding. I know the rest of the file is not well document but that doesn't mean we can't improve it, right? ;-)

Reply:
Seems like you're faster with reviewing than I am with pinging you in the issue 😆  

>  I think you move this code into the slick.core.js so that it be used by other files, 

Sure, will do that. Adapting other areas will likely be beyond the scope of this PR. 

> 1 other thing I could say is perhaps adding comments to some of the functions to explain what they do might be beneficial for future refactoring or understanding

Sure, can do that. I was trying to stick to the current code style and practices in the file so I left them out. As you might know: there are 2 kind of devs in the market: those who claim comments are bad and those who claim comments are good. 

Those who claim they are bad argue that the code should be structured and written in a way that it is clear from reading it what it is doing. 😁 e.g. some comments explaining a workflow, can be replaced by slicing the overlal method into a series of methods/functions which have proper names. 

I also rather like comments if they are meaningful.

Reply:
haha yeah I know but nowadays must developers use some kind of minification process, at least I do with WebPack, and so these comments get dropped from the prod build. If some dev aren't doing that then it's their problem, I would prefer to have more comments but yes if the methods have proper naming and are meaningful then no need to add comments, we are in agreement on that 😉